src/main.cc

```diff
@@ -474,6 +472,8 @@ int main (int argc, char *argv[])
       ReadFromUser();
       /* Check if engine input ready for reading. If so, store it in a buffer. */
       ReadFromEngine();
+
+      usleep(100);
```

Comment:
The thread calls usleep() while holding one or more locks.  This could lead to process starvation.  - usleep() can block for a long time.  - If the highlighted code executes, the following locks are held when usleep() is called at main.cc:476.      - Lock adapter::adapter_init_mutex.components.cc:89.